From a606baaa045c30060693d345ddb264abd906dd91 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Thu, 5 Oct 2017 12:30:01 +0200
Subject: [PATCH 049/146] commit 833689044e13a3fb8c6c03ad868375670d97010b
 Merge: 63a2f310d040 0ae0bb3f409d Author: Kalle Valo <kvalo@codeaurora.org>
 Date:   Mon Jun 12 19:29:39 2017 +0300
Git-commit: 63a2f310d040a4e42f92d60762c8965cd6f2a5ed
Patch-mainline: v4.13-rc1
References: FATE#322675

    First batch of iwlwifi driver patches 4.13

    * Loads of FW API documentation improvements (for tools and htmldoc);
    * Continued work for the new A000 family;
    * Bumped the maximum supported FW API to 31;
    * Improve the differentiation between 8000, 9000 and A000 families;
    * A lot of fixes and cleanups here and there;

    kvalo: There were conflicts iwl_mvm_stop_device() and
    iwl_mvm_tcool_set_cur_state(). The former was easy but latter needed more
    thought. Apparently the mutex was taken too late, so I fixed so that the mutex
    is taken first and then check for iwl_mvm_firmware_running().

oneukum: This merge was unclean. So I need to take the necessary stuff.

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/net/wireless/intel/iwlwifi/iwl-8000.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/intel/iwlwifi/iwl-8000.c
+++ b/drivers/net/wireless/intel/iwlwifi/iwl-8000.c
@@ -70,8 +70,8 @@
 #include "iwl-agn-hw.h"
 
 /* Highest firmware API version supported */
-#define IWL8000_UCODE_API_MAX	30
-#define IWL8265_UCODE_API_MAX	30
+#define IWL8000_UCODE_API_MAX	31
+#define IWL8265_UCODE_API_MAX	31
 
 /* Lowest firmware API version supported */
 #define IWL8000_UCODE_API_MIN	22
